### PR TITLE
Cleanup of config docs

### PR DESCRIPTION
**Pull request by abbyad**
Updated structure and language of the configuration documentation to reflect current use.

### CODE REVIEW

**Comment by abbyad** [comment]
I have added several reviewers because this changes how the config documentation is structured. Some of these documents were previously created/updated by developers, so I don't want anyone to feel lost. The ones that were moved to legacy had become outdated since they weren't using the declarative config style. Having them in a combined doc may seem unwieldy, but is easier to maintain consistency, flow, and context in this way.

In the near future as part of CHT we expect to revisit documentation, at which time it may get refactored again into a new documentation site. In the meantime though, I think that the combined doc and new readme will be easier to guide internal and external contributors.

File: configuration/README.md [hunk 1]
```diff
@@ -1,44 +1,53 @@
...
+- [Targets](legacy/targets.md) → Replaced by declarative Targets section in [this guide](https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#targets)
+- [Tasks](legacy/tasks.md) → Replaced by declarative Tasks section in [this guide](https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#tasks)
+- [Contact Summary](legacy/contact-summary.md) → Replaced by declarative Contacts section in [this guide](https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#contacts)
+- [Setting up XForms in V2 (MM App)](legacy/setting-up-xforms-for-mm-app.md) → Replaced by Forms section in [this guide](https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#forms)
+- [Install SSL Cerificates to VM](legacy/install-ssl-certificates-to-vm.md) → Not supported for Medic projects. Self hosting instructions to come in CHT
+- [Pyxform](legacy/pyxform.md) → Should now use [medic-conf](https://github.com/medic/medic-conf#installation)
```
Comment (derickl) [inline configuration/README.md]: medic-conf still uses pyxform.

File: configuration/legacy/pyxform.md [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-
-```
-python path/to/pyxform/xls2xform.py path_to_XLSForm output_path
-```
-
-- For example:
-
-```
-python /path/to/pyxform/xls2xform.py myExcel.xlsx myOutput.xml
+## PyxForm
```
Comment (derickl) [inline configuration/legacy/pyxform.md]: Should we be pointing users to upstream pyxform or our own fork?

File: configuration/README.md (hunk 1 above)
Comment (abbyad) [reply configuration/README.md]: The medic-conf installation guide includes the required version of pyxform

File: configuration/legacy/pyxform.md (hunk 2 above)
Comment (abbyad) [reply configuration/legacy/pyxform.md]: As per [medic-doc documentation](https://github.com/medic/medic-conf/blob/52ac649af62eaa8eee04ba47973b7e90f98b8d3c/README.md), we should be using own fork: eg `pip install git+https://github.com/medic/pyxform.git@medic-conf-1.17#egg=pyxform-medic`

Our own fork support some Medic specific types.

**Comment by abbyad** [comment]
Thanks for the reviews!

I've just merged, and moving forward it would be great for any new documentation files in the config folder to be contextualized somewhere in its README.

**Pull request merged**